crates/forge/src/running.rs

```diff
@@ -160,3 +168,32 @@ pub(crate) fn run_from_test_case(
         Err(err) => Err(err.into()),
     }
 }
+
+fn extract_fork_when_id_passed(
```

Comment:
I would separate it into two functions, one mapping Fork::Id to Fork::Config if that is the case, and then just build state from Config.

Reply:
I don't know how to do that with enums